test/common/http/conn_manager_impl_test.cc

```diff
@@ -3595,5 +3595,31 @@ TEST_F(HttpConnectionManagerImplTest, DisableKeepAliveWhenOverloaded) {
   EXPECT_EQ(1U, stats_.named_.downstream_cx_overload_disable_keepalive_.value());
 }
 
+TEST_F(HttpConnectionManagerImplTest, LargeRequestRejected) {
```

Comment:
This new test is fine, but FYI we do have coverage in an integration test also. See `HttpIntegrationTest::testOverlyLongHeaders()`. While you are here, can you name this test something more specific since it's about headers?

Reply:
o true, forgot requests can have bodies lol. updated! and thanks for the integration test ref, was just looking for that.

Reply:
(killed an unused var in the integration test too)